lib/openstack_api/openstack_server.py

```diff
@@ -4,19 +4,30 @@
 from openstack.compute.v2.image import Image
 
 
-def migrate_server(
+def snapshot_and_migrate_server(
     conn: Connection,
     server_id: str,
+    server_status: str,
+    snapshot: bool,
     dest_host: Optional[str] = None,
-    live: bool = True,
 ) -> None:
     """
-    Migrate a server to a new host
+    Optionally snapshot a server and then migrate it to a new host
     :param conn: Openstack Connection
     :param server_id: Server ID to migrate
-    :param dest_host: Optional host to migrate to, otherwise choosen by scheduler
-    :param live: True for live migration
+    :param server_status: Status of machine to migrate - must be ACTIVE or SHUTOFF
+    :param dest_host: Optional host to migrate to, otherwise chosen by scheduler
     """
+    if snapshot:
+        snapshot_server(conn=conn, server_id=server_id)
+    if server_status == "ACTIVE":
+        live = True
+    elif server_status == "SHUTOFF":
+        live = False
+    else:
+        raise ValueError(
+            f"Server status: {server_status}. The server must be ACTIVE or SHUTOFF to be migrated"
+        )
```

Comment:
here's a more concise way to write this

```python
if server_status not in ["ACTIVE", "SHUTOFF"]:
    raise ValueError(...)

if server_status == "SHUTOFF":
    return conn.compute.migrate_server(...)
return conn.compute.live_migrate_server(...)
```

saves you a few lines :) - makes it easier to read too!